src/intermediarios/validarCpf.js

```diff
@@ -0,0 +1,18 @@
+const validarCpf = (req, res, next) => {
+    let { cpf } = req.body;
+
+    cpf = cpf.replaceAll('.', '').replaceAll('-', '').trim();
+
+    if (isNaN(cpf)) {
+        return res.status(400).json({ mensagem: 'O campo CPF deve ser numérico.' });
+    }
+
+    if (cpf.length !== 11) {
+        return res.status(400).json({ mensagem: 'CPF inválido. Deve conter exatamente 11 dígitos numéricos.' });
+    }
+
```

Comment:
Como o CPF está sendo armazenado no banco de dados da mesma forma que é enviado pelo usuário, o CPF "12345678900" será considerado diferente de "123.456.789-00". Uma forma de resolver isso é alterar aqui mesmo no intermediário o campo cpf de req.body pelo cpf formatado sem os caracteres especiais. 

```suggestion
    req.body.cpf = cpf
```

Reply:
certo, corrigi agora essa parte. obrigada!